--- old/Makefile.am	2011-07-11 14:58:11.468606842 +0800
+++ new/Makefile.am	2011-07-11 15:21:42.593537802 +0800
@@ -367,6 +367,30 @@
 EXTRA_DIST += scripts/bluetooth.rules \
 		scripts/bluetooth-hid2hci.rules scripts/bluetooth-serial.rules
 
+test_scripts = test/apitest \
+	test/hsplay \
+	test/hsmicro \
+	test/list-devices \
+	test/monitor-bluetooth \
+	test/simple-agent \
+	test/simple-service \
+	test/test-adapter \
+	test/test-audio \
+	test/test-device \
+	test/test-discovery \
+	test/test-input \
+	test/test-manager \
+	test/test-network \
+	test/test-serial \
+	test/test-service \
+	test/test-telephony
+
+
+if TEST
+testdir = ${pkglibdir}/test
+test_SCRIPTS = ${test_scripts}
+endif
+
 EXTRA_DIST += doc/manager-api.txt \
 		doc/adapter-api.txt doc/device-api.txt \
 		doc/service-api.txt doc/agent-api.txt doc/attribute-api.txt \
